src/types.ts

```diff
@@ -62,17 +120,86 @@ type ExplicitContent = {
   filter_locked: boolean;
 };
 
-type ExternalUrls = {
-  spotify: string;
-};
-
 type Followers = {
   href: null;
   total: number;
 };
 
-type Image = {
-  height: null;
+export type Image = {
+  height: number;
   url: string;
-  width: null;
+  width: number;
+  dominantColor: number;
 };
+
+export type Track = {
+  album: SpotifyAlbum;
+  albumRef: DocumentReference<DocumentData>;
+  artists: Owner[];
+  disc_number: number;
+  duration_ms: number;
+  episode: boolean;
+  explicit: boolean;
+  external_ids: ExternalIDS;
+  external_urls: ExternalUrls;
+  href: string;
+  id: string;
+  is_local: boolean;
+  is_playable: boolean;
+  name: string;
+  popularity: number;
+  preview_url: string;
+  track: boolean;
+  track_number: number;
+  type: Type;
+  uri: string;
+  linked_from?: Owner;
+};
+
+export type SpotifyAlbum = {
+  album_type: string;
+  artists: Artist[];
+  external_urls: ExternalUrls;
+  href: string;
+  id: string;
+  images: Image[];
+  name: string;
+  release_date: string;
+  release_date_precision: string;
+  total_tracks: number;
+  type: string;
+  uri: string;
+};
+
+type Artist = {
+  external_urls: ExternalUrls;
+  href: string;
+  id: string;
+  name: string;
+  type: string;
+  uri: string;
+};
+
+type ExternalUrls = {
+  spotify: string;
+};
+
+type ExternalIDS = {
+  isrc: string;
+};
+
+type Owner = {
+  display_name?: string;
+  external_urls: ExternalUrls;
+  href: string;
+  id: string;
+  type: Type;
+  uri: string;
+  name?: string;
+};
+
+enum Type {
+  Artist = 'artist',
+  Track = 'track',
+  User = 'user',
+}
```

Comment:
```suggestion
type Type = 'artist' | 'track' | 'user';
```
Typeっていう型名も若干気になる
ここは完全に宗教なので任せる